services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -696,8 +652,21 @@ public long deleteMeter(final DatapathId dpid, final long meterId)
 
     @Override
     public List<Long> deleteDefaultRules(final DatapathId dpid) throws SwitchOperationException {
-        return deleteRulesWithCookie(dpid, DROP_RULE_COOKIE, VERIFICATION_BROADCAST_RULE_COOKIE,
+        IOFSwitch sw = lookupSwitch(dpid);
+
+        List<Long> deletedRules = deleteRulesWithCookie(dpid, DROP_RULE_COOKIE, VERIFICATION_BROADCAST_RULE_COOKIE,
                 VERIFICATION_UNICAST_RULE_COOKIE, DROP_VERIFICATION_LOOP_RULE_COOKIE);
+
+        try {
+            verifySwitchSupportsMeters(sw);
+
+            deleteMeter(dpid, VERIFICATION_BROADCAST_RULE_COOKIE & PACKET_IN_RULES_METERS_MASK);
+            deleteMeter(dpid, VERIFICATION_UNICAST_RULE_COOKIE & PACKET_IN_RULES_METERS_MASK);
+        } catch (UnsupportedSwitchOperationException e) {
+            logger.warn("Failed to delete default rules' meters", e);
```

Comment:
This is not valid... If switch don't support meters... on switch activation we will add meters... so we should not try to remove them on switch deactivation (or on manual request in this case)... So if switch don't support meters we should log "skip meter removal do to lack of meter support by switch dpID" on info level.